runtime/curl.cpp

```diff
@@ -172,11 +220,30 @@ class EasyContext : public BaseContext {
   string_list received_header;
   string_list received_data;
 
+  struct {
+    curl::on_write_callable callable{NULL};
+    int64_t method{KPHP_CURL_STDOUT};
+    const Stream *stream{nullptr};
+  } write_handler, write_header_handler;
+
+  struct {
+    curl::on_read_callable callable{NULL};
+    int64_t method{KPHP_CURL_DIRECT};
+    const Stream *stream{nullptr};
+  } read_handler;
+
+  struct {
+    curl::on_progress_callable callable{NULL};
+  } progress_handler;
+  
+  struct {
+    curl::on_xferinfo_callable callable{NULL};
+  } xferinfo_handler;
+
```

Comment:
## Logic for storing user-defined functions for callable options of `curl_setopt()`

In the EasyContext class was added fields, which store callable objects and additional information about the mode of operation of functions that are passed to libcurl.

Arguments information:
- `callable`: a field for user-defined function
- `method`: defines the mode of operation of the final function
- `stream`: pointer to the stream descriptor (if any)

Methods:
- `KPHP_CURL_STDOUT`: the standard output of information for `curl_write()` and `curl_write_header()`;
- `KPHP_CURL_RETURN`: a special mode of operation for `curl_write()`, in which information is returned from the function;
- `KPHP_CURL_USER`: the mode of operation in which a user-defined function is called (from `callable` field)
- `KPHP_CURL_IGNORE`: a special mode of operation for `curl_write_header()`, in which a length of the information is returned from the function;
- `KPHP_CURL_DIRECT`:  the standard output of information for `curl_read()`;
- `KPHP_CURL_FILE`:  the mode of operation for `curl_write()` and `curl_write_header()` for writing information to a stream descriptor from the `stream` field